### PR TITLE
[DataGrid] ColumnBase improve access information

### PR DESCRIPTION
**Pull request by franklupo**
To create columns outside this library I need information that isn't there now.

### CODE REVIEW

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs [hunk 1]
```diff
@@ -128,6 +128,16 @@ public abstract partial class ColumnBase<TGridItem>
...
 
+    /// <summary>
+    /// Get items.
+    /// </summary>
+    protected ICollection<TGridItem> Items => InternalGridContext.Items;
+
+    /// <summary>
+    /// Get Total Item Count.
+    /// </summary>
+    public int TotalItemCount => InternalGridContext.TotalItemCount;
```
Comment (dvoituron) [inline src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs:139]: Probably better to have a `protected´ property.

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs [hunk 2]
```diff
@@ -128,6 +128,16 @@ public abstract partial class ColumnBase<TGridItem>
     /// </summary>
     protected FluentDataGrid<TGridItem> Grid => InternalGridContext.Grid;
 
+    /// <summary>
+    /// Get items.
```
Comment (dvoituron) [inline src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs:132]: Can you give a different explanation of the property name?
`Gets a list of all elements included in the grid.`

**Review by dvoituron — CHANGES_REQUESTED** [review]

**Comment by dvoituron** [comment]
I think you can bypass these changes, adding this Cascading property in your new column code.
```csharp
[CascadingParameter]
internal InternalGridContext<TGridItem> InternalGridContext { get; set; } = default!;
```

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs (hunk 2 above)
Comment (franklupo) [reply src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs:132]: the items are calculated they are not all

**Comment by franklupo** [comment]
> I think you can bypass these changes, adding this Cascading property in your new column code.
> 
> ```cs
> [CascadingParameter]
> internal InternalGridContext<TGridItem> InternalGridContext { get; set; } = default!;
> ```

Ok but the InternalGridContext<TGridItem> Is internal

**Comment by dvoituron** [comment]
> Ok but the InternalGridContext Is internal

Indeed ;-)

**Comment by franklupo** [comment]
IMO InternalGridContext could be deleted and move properties/methods to FluentDataGrid. FluentDataGrid is found in CascadingParameter, or change visibility InternalGridContext

**Comment by dvoituron** [comment]
> IMO InternalGridContext could be deleted and move properties/methods to FluentDataGrid. FluentDataGrid is found in CascadingParameter, or change visibility InternalGridContext

Nop. `InternalGridContext` exists to centralize some internal properties. The next major release could refactor this part to simplify the extensions.

**Comment by vnbaaij** [comment]
> IMO InternalGridContext could be deleted and move properties/methods to FluentDataGrid. FluentDataGrid is found in CascadingParameter, or change visibility InternalGridContext

The `InternalGridContext` should be seen as the object holding the grid's internal state. We do _not_ want to expose that.

**Comment by franklupo** [comment]
the items comes from here and it is not all the items

https://github.com/microsoft/fluentui-blazor/blob/cfb055ced298d26f83ebcb21ed31a636025db63c/src/Core/Components/DataGrid/FluentDataGrid.razor.cs#L512

File: src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs (hunk 2 above)
Comment (franklupo) [reply src/Core/Components/DataGrid/Columns/ColumnBase.razor.cs:132]: the items comes from here and it is not all the items

https://github.com/microsoft/fluentui-blazor/blob/cfb055ced298d26f83ebcb21ed31a636025db63c/src/Core/Components/DataGrid/FluentDataGrid.razor.cs#L512

**Comment by franklupo** [comment]
something is missing?

**Comment by franklupo** [comment]
Are there any problems applying this change?

**Comment by vnbaaij** [comment]
Hi Daniele, still not have time to take a proper look at this and won't be able for the coming weeks because of vacation period. There is no rush to get this in as it will require at least a minor release (i.e. 4.X)

**Comment by franklupo** [comment]
you're right, this change gives me the ability to create components outside of this library.

**Comment by vnbaaij** [comment]
Closing this as there has been no activity for a while.

**Pull request closed without merging**